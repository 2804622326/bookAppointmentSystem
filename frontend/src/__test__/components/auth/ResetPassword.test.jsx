import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import ResetPassword from '../../../components/auth/ResetPassword.jsx';

jest.mock('../../../components/auth/AuthService.js', () => ({
  validateToken: jest.fn(() => Promise.resolve({ message: 'VALID' })),
  resetPassword: jest.fn(() => Promise.resolve({ message: 'ok' })),
}));

jest.mock('../../../components/common/ProcessSpinner.jsx', () => () => <div data-testid='spinner' />);

jest.mock('../../../components/hooks/UseMessageAlerts', () => ({ __esModule: true, default: jest.fn() }));
import useAlerts from '../../../components/hooks/UseMessageAlerts';
const setErrorMessage = jest.fn();
const setSuccessMessage = jest.fn();
const setShowSuccessAlert = jest.fn();
const setShowErrorAlert = jest.fn();
useAlerts.mockReturnValue({
  errorMessage: '',
  successMessage: '',
  showErrorAlert: false,
  showSuccessAlert: false,
  setErrorMessage,
  setSuccessMessage,
  setShowSuccessAlert,
  setShowErrorAlert,
});

beforeEach(() => {
  window.history.pushState({}, '', '/reset?token=abc');
});

test('submits new password after validation', async () => {
  const { asFragment } = render(<ResetPassword />);
  await waitFor(() => screen.getByText(/reset your password/i));
  fireEvent.change(screen.getByPlaceholderText(/choose a new password/i), { target: { value: 'newpass' } });
  fireEvent.click(screen.getByRole('button', { name: /reset password/i }));
  const { resetPassword } = require('../../../components/auth/AuthService.js');
  await waitFor(() => expect(resetPassword).toHaveBeenCalledWith('abc', 'newpass'));
  expect(setShowSuccessAlert).toHaveBeenCalledWith(true);
  expect(asFragment()).toMatchSnapshot();
});
