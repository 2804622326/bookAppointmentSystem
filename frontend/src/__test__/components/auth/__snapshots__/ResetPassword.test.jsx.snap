// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submits new password after validation 1`] = `
<DocumentFragment>
  <div
    class="d-flex align-items-center justify-content-center container"
    style="margin-top: 100px;"
  >
    <div
      class="w-100 card"
      style="max-width: 600px;"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        >
          Reset Your Password
        </div>
        <form
          class=""
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="emailInput"
            >
              Set a new password
            </label>
            <input
              class="form-control"
              id="emailInput"
              placeholder="choose a new password"
              type="password"
              value="newpass"
            />
          </div>
          <button
            class="btn btn-outline-info"
            type="submit"
          >
            Reset Password
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
